src/App.tsx

```diff
@@ -15,36 +17,83 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SORT_BY_ALPH: number = 1;
+const SORT_BY_LENGTH: number = 2;
+
+const sortGoods = (goods: string[], key: number): string[] => {
+  switch (key) {
+    case SORT_BY_ALPH:
+      return goods.sort((a, b) => a.localeCompare(b));
+    case SORT_BY_LENGTH:
+      return goods.sort((a, b) => a.length - b.length);
+    default:
+      return goods;
+  }
+};
+
 export const App: React.FC = () => {
```

Comment:
[CODE KNOWLEDGE] The App component is directly interacting with the DOM by sorting and reversing the goods list. This is not recommended in React. Instead, you should keep track of the sorting and reversing state in your component, and use this state when rendering the list of goods.